### PR TITLE
BFD-2928: Convert remaining EOB V2 E2E into RestAssured

### PR DESCRIPTION
**Pull request by lsmitchell**
**JIRA Ticket:**
[BFD-2928](https://jira.cms.gov/browse/BFD-2928)

**User Story or Bug Summary:**

As a BFD Engineer I would like the existing resource provider E2E tests to use restAssured so that we are testing our REST endpoints as customers use them.

---

### What Does This PR Do?

A majority of the tests inside of `R4ExplanationOfBenefitResourceProviderE2E.java` were converted into RestAssured tests and placed into ExplanationOfBenefitE2E as part of the hackathon work. This PR converts remaining straggler tests and test cases not captured in that initial work, with the intention of completely replacing R4ExplanationOfBenefitResourceProviderE2E with RestAssured versions of the tests.

This story scope is specifically the v2 versions of the EOB E2E.

- Fixes the (v2) ExplanationOfBenefitE2E samhsa tests to cover all claim types as they were covered in the original E2E
    - Moves the samhsa adding logic into the common server utility class
- Adds tests to cover all claim types for the read endpoint, and extracts common logic to make the tests DRY
- Adjust the pagination test to use an odd pagination value to cover a test case from the original E2E
    - Test case is where we have an odd pagination number (and an even number of results) we make sure the last 2 items dont cause an out-of-bounds exception
- Adds missing restAssured test for pagination if only startIndex is present
- Added samhsa test to ensure default behavior for when samhsa flag is not explicitly added
- Added some test utility method adjustments to cover all claim types instead of just carrier
- Deleted R4ExplanationOfBenefitResourceProviderE2E since all test cases are now covered in ExplanationOfBenefitE2E

One omission is the field comparison tests that existed in R4ExplanationOfBenefitResourceProviderE2E. These tests copy the same case as EndpointJsonResponseComparatorV2IT, as they just load the SampleA data and run it through the resource provider+transformer, then compare the output to the original loaded data. The EndpointJson tests do the same thing (but with the final json output) so these tests are superfluous.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.
* Check if theres anything I missed about the field-checking tests removed from the deleted E2E where they are not just covered by the endpointJson tests

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/ExplanationOfBenefitE2E.java [hunk 1]
```diff
@@ -53,42 +59,87 @@ public void setupTest() {
...
   }
 
   /** Verifies that the EOB returns a 400 when called with a negative count parameter. */
   @Test
   public void testEobByPatientIdWithNegativeCountExpect400() {
 
     String patientId = testUtils.getPatientId(testUtils.loadSampleAData());
-    // It's also possible to add these to the call below, but I liked the request string as
-    // one would write it when calling the API
+    // Its possible to add these via the API, but
```
Comment (mackec1) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/ExplanationOfBenefitE2E.java:142]: seems a little wonky....", but" what?

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/ExplanationOfBenefitE2E.java [hunk 2]
```diff
@@ -926,44 +1051,89 @@ public void searchEobByPatientIdWithNoResultsAndPaginationRequestedExpect200() {
...
-    return loadedRecords.stream()
-        .filter(r -> r instanceof OutpatientClaim)
-        .map(OutpatientClaim.class::cast)
-        .findFirst()
-        .get();
+  private String getClaimIdFor(List<Object> loadedRecords, ClaimType claimType) {
+    return switch (claimType) {
+      case CARRIER:
+        CarrierClaim carrier = getClaim(loadedRecords, CarrierClaim.class);
+        yield String.valueOf(carrier.getClaimId());
```
Comment (mackec1) [inline apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/ExplanationOfBenefitE2E.java:1081]: NICE! I like this use of yield.

**Review by mackec1 — DISMISSED** [review]
one dangling comment

File: apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/ExplanationOfBenefitE2E.java (hunk 1 above)
Comment (lsmitchell) [reply apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/ExplanationOfBenefitE2E.java]: Oops, had a fix for this and never pushed it. Fixed!

**Review by brandoncruz3 — APPROVED** [review]

**Review by mackec1 — APPROVED** [review]

**Pull request merged**